orbstation/spells/eldritch_minion.dm

```diff
@@ -0,0 +1,230 @@
+	/// What are we going to summon?
+	var/datum/demonological_summon/chosen_minion
+
+/obj/effect/demonology_rune/can_interact(mob/living/user)
+	. = ..()
+	if(!.)
+		return
+	if(is_in_use)
+		return FALSE
+
+/obj/effect/demonology_rune/interact(mob/living/user)
+	. = ..()
+	if (!chosen_minion)
+		choose_minion(user)
+		return
+
+	INVOKE_ASYNC(src, .proc/summon_minion, user)
+	return TRUE
+
+/// Pick a minion
+/obj/effect/demonology_rune/proc/choose_minion(mob/living/user)
+	var/static/list/options = list()
+	var/static/list/picks_to_instances = list()
+
+	if (length(options) == 0)
+		for (var/type as anything in subtypesof(/datum/demonological_summon))
+			var/datum/demonological_summon/summon = new type()
+			var/datum/radial_menu_choice/choice = summon.generate_radial_entry()
+			if (!choice)
+				continue
+			options += list("[choice.name]" = choice)
+			picks_to_instances[choice.name] = summon
+
+	var/pick = show_radial_menu(user, user, options, require_near = TRUE, tooltips = TRUE)
+	if (!pick)
+		return
+	chosen_minion = picks_to_instances[pick]
+
+/// Try to make a minion, if ghosts are up for it
+/obj/effect/demonology_rune/proc/summon_minion(mob/living/user)
+	is_in_use = TRUE
+	var/mob/summoned = chosen_minion.summon(user, get_turf(src))
+	is_in_use = FALSE
+
+	if (!summoned)
+		return
+
+	user.visible_message(span_notice("[src] burns away in a flash as the [summoned.name] corporealises!"))
+	new /obj/effect/decal/cleanable/ash(get_turf(user))
+	qdel(src)
+
+/**
+ * That's right maintainers it's three o'clock, time for another of Jac's datums to use "subtypesof" on.
+ * This one represents a beastie you can make. Frankly, I only really need it for the tooltips.
+ * Unfortunately I think the tooltips are important.
+ */
+/datum/demonological_summon
+	/// Description of the beast
+	var/desc
+	/// Path of the beast
+	var/mob_path
+
+/datum/demonological_summon/proc/generate_radial_entry()
+	if (!mob_path)
+		return
+	var/datum/radial_menu_choice/choice = new()
+	var/mob/demon_type = mob_path
+	choice.name = initial(demon_type.name)
+	choice.image = image(icon = initial(demon_type.icon), icon_state = initial(demon_type.icon_state))
+	choice.info = desc
+	return choice
+
+/**
+ * Create a monster and try and put a ghost inside it
+ * This is mostly just copied heretic code, if it ain't broke...
+ */
+/datum/demonological_summon/proc/summon(mob/living/user, turf/summon_location)
+	var/mob/living/summoned = new mob_path(summon_location)
+	summoned.alpha = 0
+	summoned.notransform = TRUE
+	summoned.move_resist = MOVE_FORCE_OVERPOWERING
+	animate(summoned, 10 SECONDS, alpha = 155)
+
+	message_admins("A [summoned.name] is being summoned by [ADMIN_LOOKUPFLW(user)] in [ADMIN_COORDJMP(summoned)].")
+	var/list/mob/dead/observer/candidates = poll_candidates_for_mob("Do you want to play as a [summoned.real_name]?", ROLE_HERETIC, FALSE, 10 SECONDS, summoned)
+	if(!LAZYLEN(candidates))
+		summon_location.balloon_alert(user, "ritual failed, no ghosts!")
```

Comment:
IMO it makes more sense for this to be in demonology_rune/summon_minion(), since that's where the success message is. It's easier to read if similar things are put in similar spots. But that's a nitpick, and again might just be from copying over heretic code.